backend/apollo/elections/serializers.py

```diff
@@ -55,17 +59,53 @@ def authorize(self, attrs):
 
     answer = serializers.PrimaryKeyRelatedField(queryset=Answer.objects.all())
     author = serializers.HiddenField(default=CurrentUserDefault())
+    election = serializers.PrimaryKeyRelatedField(queryset=Election.objects.all(), write_only=True)
 
     class Meta:
         model = Vote
-        fields = ["id", "answer", "author"]
+        fields = ["id", "answer", "author", "election"]
         read_only_fields = ["id"]
+        write_only_fields = ["election"]
         validators = [
             UniqueTogetherValidator(
                 queryset=Vote.objects.all(), fields=["author", "answer"]
             )
         ]
 
+    def is_valid(self, raise_exception=False):
+        try:
+            answer_id_ciphertext = self.initial_data["answer"]
+            election_id = self.initial_data["election"]
+        except KeyError:
+            if raise_exception:
+                raise ValidationError()
+            return False
+
+        election = get_object_or_404(Election, pk=election_id)
+
+        try:
+            answer_id = decrypt(election.secret_key, answer_id_ciphertext)
+        except CryptoError:
+            if raise_exception:
+                raise ValidationError()
+            return False
+
+        answer = get_object_or_404(Answer, pk=answer_id)
+
+        if answer.question.election.id != election_id:
+            if raise_exception:
+                raise ValidationError()
+            return False
+
+        self.initial_data["answer"] = answer_id
+
+        return super().is_valid(raise_exception)
```

Comment:
How about this?

```python
    def _decrypt_election(self):
        election = get_object_or_404(Election, pk=self.initial_data.pop("election"))
        answer_id = decrypt(election.secret_key, self.initial_data["answer"])
        answer = get_object_or_404(Answer, pk=answer_id)

        if answer.question.election.id != election_id:
            raise ValidationError()

        return answer_id

    def is_valid(self, raise_exception):
        try:
            answer_id = self._decrypt_election()
        except (KeyError, CryptoError, ValidationError):
            if raise_exception:
                raise ValidationError()

        self.initial_data["answer"] = answer_id
        return super().is_valid(raise_exception)

```
With `pop` in the first line you don't need the `create` method.

In the future we can think about doing it even prettier.

Reply:
That is much better, thanks